### PR TITLE
fix: optional project bigquery

### PR DESCRIPTION
**Pull request by rephus**
Closes:  #9515

### Description:

This optional project is only valid for oauth method.

New error for non oauth: 

Screenshot from 2024-04-02 08-57-29

Oauth without project: 

image

image

A new project was created pointing to our default org in gcloud , so it is expected to throw an error for jaffle shop when running queries.

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/Explorer/CustomMetricModal/utils/index.ts [hunk 1]
```diff
@@ -166,6 +181,7 @@ export const prepareCustomMetricData = ({
         filters: customMetricFilters.length > 0 ? customMetricFilters : [],
         label: customMetricLabel,
         name: getCustomMetricName(
+            item.table,
```
Comment (ZeRego) [inline packages/frontend/src/components/Explorer/CustomMetricModal/utils/index.ts:184]: Are these changes related?

Comment (rephus) [reply packages/frontend/src/components/Explorer/CustomMetricModal/utils/index.ts:184]: no sorry, I commited this by accident

**Review by ZeRego — APPROVED** [review]
